design/cluster-status-conditions.md

```diff
@@ -0,0 +1,39 @@
+# Cluster Status Condition
+
+## Background
+
+Currently, Ceph Cluster stores the state of the system in `Status.State`.However OLM expects the state to be in `Status.Phase`. The elimination of State variable breaks the backward compatibility of the Ceph Cluster. Although keeping both of them is also not a good practice. But the usage of `Status.Phase` is deprecated over time because it contradicts the system design principle and hampered evolution. So rather than encouraging clients to infer the implicit properties from phases, the usage of `Status.Condition` is preferred.
+
+## Conditions
+
+Conditions simply represent the latest available observation of an object's state. They are an extension mechanism intended to be used when the details of observation are not a priori known or would not apply to all instances of a given Kind. Objects can have multiple Conditions, and new types of Conditions can also be added in the future by the third-party controllers. Thus, Conditions are thereby represented using list/slice, where each having the similar structure.
+
+## System States for rook-ceph
+
+The necessary system states for the rook-ceph can be portrayed as follows:
+	   
+	   Ignored 	: If any of the resources gets ignored for multiple reasons.
+	   Progressing 	: Marks the start of reconcile of Ceph Cluster
+	   Ready 	: When Reconcile completes successfully
+	   Not Ready 	: Either when cluster is Upgrading or Upgrading is blocked
+	   Available 	: The Ceph Cluster is healthy and is available to use.
+	   Failure 	: If any failure occurs in the Ceph Cluster
+
+## Implementation Details
+
+Since the use of Phases are deprecated it is better to use the Conditions to describe the system states of the rook-ceph. 
```

Comment:
This statement adds no value here. It should be part of the `Background`.

Reply:
Okay